pinot-core/src/main/java/org/apache/pinot/server/realtime/ControllerLeaderLocator.java

```diff
@@ -214,6 +191,9 @@ private boolean isLeadControllerResourceEnabled() {
    * @param instanceId instance id without any prefix, e.g. localhost_9000
    * */
   private Pair<String, Integer> convertToHostAndPortPair(String instanceId) {
+    if (instanceId == null) {
```

Comment:
ok, we are going to propagate exception  handling further I suppose. For now, this is ok to keep us from changing all code in one go. Maybe add a todo

Reply:
TODO added.